docs/decisions/0009-dotnet-project-structure.md

```diff
@@ -0,0 +1,289 @@
+- Namespaces are abbreviated to remove Microsoft.SemanticKernel prefix. Namespaces use `_` rather than `.`.
+
+<img src="./diagrams/skfunctions-preview.png" alt="ISKFunction class relationships" width="400"/>
+
+## Decision Drivers
+
+- Avoid having too many assemblies because of impact of signing these and to reduce complexity
+- Follow .Net naming guidelines
+  - [Names of Assemblies and DLLs](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-assemblies-and-dlls)
+  - [Names of Namespaces](https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/names-of-namespaces)
+
+## Considered Options
+
+- Option #1: New `planning`, `functions` and `plugins` project areas
+- Option #2: Folder naming matches assembly name
+- … <!-- numbers of options can vary -->
+
+In all cases the following changes will be made:
+
+- Move non core Connectors to a separate repository
+- Merge prompt template engine and semantic functions into a single package
+
+## Decision Outcome
+
+Chosen option: "{title of option 1}", because
+{justification. e.g., only option, which meets k.o. criterion decision driver | which resolves force {force} | … | comes out best (see below)}.
+
+Main categories for the projects will be:
+
+1. `Connectors`: ***A connector project allows the Semantic Kernel to connect to AI and Memory services***. Some of the existing connector projects may move to other repositories.
+1. `Planners`: ***A planner project provides one or more planner implementations which take an ask and convert it into an executable plan to achieve that ask***. This category will include the current action, sequential and stepwise planners (these could be merged into a single project). Additional planning implementations e.g., planners that generate Powershell or Python code can be added as separate projects.
+1. `Functions`: ***A function project that enables the Semantic Kernel to access the functions it will orchestrate***. This category will include:
+    1. Semantic functions i.e., prompts executed against an LLM
+    1. GRPC remote procedures i.e., procedures executed remotely using the GRPC framework
+    1. Open API endpoints i.e., REST endpoints that have Open API definitions executed remotely using the HTTP protocol
+1. `Plugins`: ***A plugin project contains the implementation(s) of a Semantic Kernel plugin***. A Semantic Kernel plugin is contains a concrete implementation of a function e.g., a plugin may include code for basic text operations.
+
+### Option #1: New `planning`, `functions` and `plugins` project areas
+
+```text
+SK-dotnet
+├── samples/
+└── src/
+    ├── connectors/
+    │   ├── Connectors.AI.OpenAI*
+    │   ├── Connectors.AI.HuggingFace
+    │   ├── Connectors.Memory.AzureCognitiveSearch
+    │   ├── Connectors.Memory.Qdrant
+    │   ├── ...
+    │   └── Connectors.UnitTests
+    ├── planners/
+    │   ├── Planners.Action*
+    │   ├── Planners.Sequential*
+    │   └── Planners.Stepwise*
```

Comment:
nit: need unit tests project for Planners

Reply:
In the final proposal unit tests become a subfolder under the Assembly folder. We aren't there yet but this is the direction we want to go.